### PR TITLE
chore(`blob-explorers`): accomodate new blobscan API changes

### PR DESCRIPTION
**Pull request by Evalir**
Blobscan had some more API changes. They added a few things:
- a field for tx category, present on both block/tx responses.
- added the rollup field for the block response as well.
- They renamed the `DataStorageReferences` fields to be `storage` and `url`.

This aligns the new responses to the new format, and another tx test. [See their API docs](https://api.blobscan.com/#/blocks/block-getByBlockId). Although I saw https://github.com/foundry-rs/block-explorers/pull/66 open, the behavior implemented in that PR doesn't seem up to date anymore.

### CODE REVIEW

File: crates/blob-explorers/src/response.rs [hunk 1]
```diff
@@ -50,6 +50,10 @@ pub struct FullTransactionDetails {
     pub from: Address,
     /// Receiver address,
     pub to: Address,
+    /// The category of this tx, if any.
+    pub cateogory: Option<String>,
+    /// The rollup this tx corresponds to, if any.
+    pub rollup: Option<String>,
```
Comment (klkvr) [inline crates/blob-explorers/src/response.rs:56]:
```suggestion
    /// The category of this tx, if any.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub category: Option<String>,
    /// The rollup this tx corresponds to, if any.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub rollup: Option<String>,
```

File: crates/blob-explorers/src/response.rs [hunk 2]
```diff
@@ -163,10 +173,6 @@ impl TransactionDetails {
 #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct BlockInfo {
-    /// Block hash.
-    pub hash: B256,
-    /// Block timestamp.
-    pub timestamp: chrono::DateTime<chrono::offset::Utc>,
```
Comment (mattsse) [inline crates/blob-explorers/src/response.rs:169]: aren't those still available?
feels odd that those are removed

image

File: crates/blob-explorers/src/response.rs [hunk 3]
```diff
@@ -113,19 +117,25 @@ pub struct BlobInfo {
...
 pub struct TransactionDetails {
+    /// The hash of this tx.
     pub hash: B256,
     pub block_number: u64,
+    /// The hash of the block this tx is contained in.
     pub block_hash: B256,
+    /// The category of this tx, if any
+    pub category: Option<String>,
+    /// The rollup this tx corresponds to, if any
     pub rollup: Option<String>,
```
Comment (klkvr) [inline crates/blob-explorers/src/response.rs:137]:
```suggestion
    /// The category of this tx, if any
    #[serde(skip_serializing_if = "Option::is_none")]
    pub category: Option<String>,
    /// The rollup this tx corresponds to, if any
    #[serde(skip_serializing_if = "Option::is_none")]
    pub rollup: Option<String>,
```

**Review by klkvr — CHANGES_REQUESTED** [review]

File: crates/blob-explorers/src/response.rs (hunk 2 above)
Comment (Evalir) [reply crates/blob-explorers/src/response.rs:169]: These fields are present when fetching a block, for which `BlockResponse<T>` has these fields—`BlockInfo` doesn't need them.

**Comment by Evalir** [comment]
@mattsse @klkvr addressed (wanted to commit your patches directly but github was complaining about an updated diff, had to commit myself)

**Review by klkvr — APPROVED** [review]
lgtm, pending @mattsse

**Pull request merged**